src/rust/sdk/src/view.rs

```diff
@@ -0,0 +1,1128 @@
+            None => None,
+        })
+    }
+
+    /// Call `get_open_orders` view function.
+    ///
+    /// Arguments:
+    /// * `market_id`: the id of the market to get the orders from.
+    /// * `n_asks_max`: the maximum number of asks to index.
+    /// * `n_bids_max`: the maximum number of bids to index.
+    pub async fn get_open_orders(
+        &self,
+        market_id: u64,
+        n_asks_max: u64,
+        n_bids_max: u64,
+    ) -> EconiaResult<OrdersView> {
+        let managed_coin = format!("{}::market", self.econia_address);
+        let module = MoveModuleId::from_str(&managed_coin)
+            .map_err(|a| EconiaError::InvalidModuleId(a.to_string()))?;
+        let name = IdentifierWrapper::from_str("get_open_orders").unwrap();
+        let response = self
+            .client
+            .view(
+                &ViewRequest {
+                    function: aptos_api_types::EntryFunctionId { module, name },
+                    type_arguments: vec![],
+                    arguments: vec![
+                        json!(market_id.to_string()),
+                        json!(n_asks_max.to_string()),
+                        json!(n_bids_max.to_string()),
+                    ],
+                },
+                None,
+            )
+            .await?;
+        let value: OrdersView = serde_json::from_value(
+            response
+                .inner()
+                .get(0)
+                .ok_or(EconiaError::InvalidResponse)?
+                .clone(),
+        )?;
+        Ok(value)
+    }
+
+    /// Call `get_open_orders_all` view function.
+    ///
+    /// Arguments:
+    /// * `market_id`: the id of the market to get the orders from.
+    pub async fn get_open_orders_all(&self, market_id: u64) -> EconiaResult<OrdersView> {
+        let managed_coin = format!("{}::market", self.econia_address);
+        let module = MoveModuleId::from_str(&managed_coin)
+            .map_err(|a| EconiaError::InvalidModuleId(a.to_string()))?;
+        let name = IdentifierWrapper::from_str("get_open_orders_all").unwrap();
+        let response = self
+            .client
+            .view(
+                &ViewRequest {
+                    function: aptos_api_types::EntryFunctionId { module, name },
+                    type_arguments: vec![],
+                    arguments: vec![json!(market_id.to_string())],
+                },
+                None,
+            )
+            .await?;
+        let value: OrdersView = serde_json::from_value(
+            response
+                .inner()
+                .get(0)
+                .ok_or(EconiaError::InvalidResponse)?
+                .clone(),
+        )?;
+        Ok(value)
+    }
+
+    /// Call `get_posted_order_id_side` view function.
+    ///
+    /// Arguments:
+    /// * `user`: the address of the user to check market account ids for.
+    /// * `market_id`: the id of the market to check market accounts for.
```

Comment:
I'll note that this function also reduces to bitwise operations that can be computed off-chain (only because you were doing bitwise rather than chain calls for other such view funcs), not that you need to update it to the bitwise operations

Reply:
I missed that. I saw the call to some functions in it and thought it was something more complex. I did it in Rust as well now.